lntest/itest/lnd_test.go

```diff
@@ -1659,6 +1679,47 @@ func testUpdateChannelPolicy(net *lntest.NetworkHarness, t *harnessTest) {
 		t.Fatalf("unable to send payment: %v", err)
 	}
 
+	// Check that Carol has internalized the max HTLC portion of the update by
+	// attempting to send a payment 1 satoshi over the maximum, which should
+	// fail.
+	payAmt = lnwire.MilliSatoshi(maxHtlc).ToSatoshis() + 1
+	invoice = &lnrpc.Invoice{
+		Memo:  "testing",
+		Value: int64(payAmt),
+	}
+	resp, err = net.Alice.AddInvoice(ctxb, invoice)
+	if err != nil {
+		t.Fatalf("unable to add invoice: %v", err)
+	}
+
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	err = completePaymentRequests(
+		ctxt, carol, []string{resp.PaymentRequest}, true,
+	)
+
+	// Carol knows about the channel policy of Bob and should therefore
+	// not be able to find a path during routing.
+	if err == nil ||
+		!strings.Contains(err.Error(), "unable to find a path") {
+		t.Fatalf("expected payment to fail, instead got %v", err)
+	}
+
+	// Ensure that we can't update a channel's max HTLC with a value above the
+	// max in-flight msats set by the peer.
+	req.MaxHtlcMsat = calculateMaxHtlc(lnd.MaxBtcFundingAmount) + 1
+	_, err = net.Bob.UpdateChannelPolicy(ctxb, req)
+	if err == nil || !strings.Contains(err.Error(), "is above max pending") {
+		t.Fatalf("expected policy update to fail, instead got %v", err)
+	}
+
+	// Ensure that we can't update a channel's max HTLC with a value below the
+	// min HTLC.
+	req.MaxHtlcMsat = defaultMinHtlc - 1
+	_, err = net.Bob.UpdateChannelPolicy(ctxb, req)
+	if err == nil || !strings.Contains(err.Error(), "is below min HTLC") {
+		t.Fatalf("expected policy update to fail, instead got %v", err)
+	}
```

Comment:
I am not sure if it is justified to cover this in an integration test. Integration tests come at the cost of build/test time on travis and additional maintenance cost. To me this looks like it could just as well be tested in a unit test?

Reply:
Hm, yeah. I removed the parts that test incorrect `MaxHtlc`'s.